src/cdk/a11y/focus-trap/configurable-focus-trap.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {NgZone} from '@angular/core';
+import {InteractivityChecker} from '../interactivity-checker/interactivity-checker';
+import {FocusTrap} from './focus-trap';
+import {FocusTrapManager, ManagedFocusTrap} from './focus-trap-manager';
+import {FocusTrapInertStrategy} from './focus-trap-inert-strategy';
+
+/**
+ * Class that allows for trapping focus within a DOM element.
+ *
+ * This class uses a strategy pattern that determines how it traps focus.
+ * See FocusTrapInertStrategy.
+ */
+export class ConfigurableFocusTrap extends FocusTrap implements ManagedFocusTrap {
+  /** Whether the FocusTrap is enabled. */
+  get enabled(): boolean { return this._enabled; }
```

Comment:
We can! But since we aren't changing the cdkTrapFocus API, would you rather have it be inconsistent with the rest of the library, or inconsistent with cdkTrapFocus? If I make it `disabled`, cdkTrapFocus will have code like `get enabled(): boolean { return !this.focusTrap.disabled; }`.

Reply:
Ah, I forgot that the selector itself is the `enabled` property (e.g. `[cdkTrapFocus]="isTrapped"`), so doing the same thing should be fine here.

Reply:
Cool! I'll leave it then.